packages/python/plotly/plotly/express/_imshow.py

```diff
@@ -137,11 +153,33 @@ def imshow(
 
     In order to update and customize the returned figure, use
     `go.Figure.update_traces` or `go.Figure.update_layout`.
+
+    If an xarray is passed, dimensions names and coordinates are used for
+    axes labels and ticks.
     """
     args = locals()
     apply_default_cascade(args)
+    img_is_xarray = False
+    if xarray_imported:
+        if isinstance(img, xarray.DataArray):
+            y_label, x_label = img.dims[0], img.dims[1]
+            # np.datetime64 is not handled correctly by go.Heatmap
+            for ax in [x_label, y_label]:
+                if np.issubdtype(img.coords[ax].dtype, np.datetime64):
+                    img.coords[ax] = img.coords[ax].astype(str)
+            x = img.coords[x_label]
+            y = img.coords[y_label]
+            img_is_xarray = True
+            if aspect is None:
+                aspect = "auto"
+            z_name = img.attrs["long_name"] if "long_name" in img.attrs else "z"
```

Comment:
someday maybe we allow this to be overridden somehow? maybe someone likes their `long_name` the way it is but wants to have something else? Like in the `air` example above, the built-in `long_name` is `4xDaily Air temperature at sigma level 995` but the `var_desc` is `Air Temperature` ...

Reply:
OK I see, this is what they use in http://xarray.pydata.org/en/stable/plotting.html ... works for me for now I think! maybe we can eventually use `labels` to override this